fdbserver/VersionedBTree.actor.cpp

```diff
@@ -6465,9 +6413,14 @@ class KeyValueStoreRedwoodUnversioned : public IKeyValueStore {
 
 		wait(cur.seekGTE(key, 0));
 		if (cur.isValid() && cur.get().key == key) {
-			Value v = cur.get().value.get();
-			int len = std::min(v.size(), maxLength);
-			return Value(v.substr(0, len));
+			// Return a Value whose arena is the source page's arena
+			Value v;
+			v.arena().dependsOn(cur.back().page->getArena());
+			v.contents() = cur.get().value.get();
+			if (v.size() > maxLength) {
+				v.contents() = v.substr(0, maxLength);
```

Comment:
Is this check the only code difference between readValuePrefix_impl and readValue_impl? If so, couldn't we simplify the code slightly by making readValue call readValuePrefix_impl with the VALUE_SIZE_LIMIT knob as the value size (or 2^32-1)?
It doesn't seem like the performance hit would be meaningful.

Reply:
It is, and yeah I will combine these somehow.  The point of having separate APIs was, I think, to avoid copying large values if only a prefix is desired, but with now with zero-copy reads the common case, `readValue` could just return the entire value and `readValuePrefix` can use that result and truncate it.